client_android/app/src/main/java/com/securecall/app/ui/SettingsFragment.kt

```
@@ -262,6 +265,34 @@ class SettingsFragment : PreferenceFragmentCompat() {
+    }
+
+    private fun setupCrashReportPreference() {
+        findPreference<SwitchPreferenceCompat>("pref_crash_reports")?.apply {
+            isVisible = BuildConfig.TIER == "FREE"
+            setOnPreferenceChangeListener { _, newValue ->
+                val enabled = newValue as Boolean
+                try {
+                    val crashlytics = com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance()
+                    crashlytics.setCrashlyticsCollectionEnabled(enabled)
+                    if (!enabled) crashlytics.deleteUnsentReports()
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)